src/main/scala/mesosphere/marathon/api/akkahttp/v2/AppsController.scala

```diff
@@ -262,7 +265,33 @@ class AppsController(
       }
   }
 
-  private def deleteSingle(appId: PathId)(implicit identity: Identity): Route = ???
+  private def deleteSingle(appId: PathId)(implicit identity: Identity): Route =
+    forceParameter { force =>
+      onSuccess(groupManager.updateRootEither(appId.parent, deleteAppRootGroupModifier(appId), force = force)) {
+        case Right(plan) =>
+          completeWithDeploymentForApp(appId, plan)
+        case Left(rej) =>
+          reject(rej)
+      }
+    }
+
+  /**
+    * We have to pass this function to the groupManager to make sure updates are serialized
```

Comment:
Hm, I think we need even more details.

```
We have to pass this function to the groupManager to make sure updates are serialized
	
Another request might remove the app before we call the update. That is why we cannot check if the app exists before the update call. We have to do so during the call. Since we do some app handling anyways we can also check if the caller is authroized.
```